controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -180,37 +217,77 @@ public String getName() {
+                                    List<String> streamTagsInScopeTableNames = streamTagsInScopeTableNameFuture.join();
+                                    val deleteTagTablesFut = streamTagsInScopeTableNames.stream()
+                                                                                        .map(tableName -> storeHelper.deleteTable(tableName, false, context.getRequestId()))
+                                                                                        .collect(Collectors.toList());
+                                    return CompletableFuture.allOf(storeHelper.deleteTable(streamsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(kvtsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(rgsInScopeTableName, true, context.getRequestId()),
+                                                                   Futures.allOf(deleteTagTablesFut))
+                                                            .thenAccept(v -> log.debug("tables deleted {} {} {}", streamsInScopeTableName,
+                                                                                       kvtsInScopeTableName, rgsInScopeTableName));
+                                })
+                                .thenCompose(deleted -> storeHelper.removeEntry(SCOPES_TABLE, scopeName, context.getRequestId()));
     }
 
     @Override
-    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor, 
+    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor,
                                                                      OperationContext context) {
         Preconditions.checkNotNull(context, "Operation context cannot be null");
         return getStreamsInScopeTableName(context)
                 .thenCompose(streamsInScopeTable -> readAll(limit, continuationToken, streamsInScopeTable, context));
     }
 
+    @Override
+    public CompletableFuture<Pair<List<String>, String>> listStreamsForTag(String tag, String continuationToken, Executor executor,
+                                                                           OperationContext context) {
+
+        Preconditions.checkNotNull(context, "Operation context cannot be null");
+
+        return getStreamsFromNextTagChunk(tag, continuationToken, context).thenCompose(pair -> {
+            if (pair.getLeft().isEmpty() && !pair.getRight().contains(LAST_TAG_CHUNK)) {
+                return listStreamsForTag(tag, pair.getRight(), executor, context);
+            } else {
+                return CompletableFuture.completedFuture(pair);
+            }
+        });
+    }
+
+    /*
+        Fetch the Streams from the next available Tag Chunk table.
```

Comment:
i thought all chunks of entry for the same tag are in the same table segment.. 
remind me of the model and nomenclature again please:

1. streams corresponding to a tag are all written in one table segment based on the hash of the tag, right?
2. streams for a tag are chunked across multiple rows, right?

so if this method takes a "tag" as param, i would presume its looking for all streams corresponding to that tag. 
and that all can be found in a single chunk.

Reply:
I have improved the java docs. Could you check it now ?